### PR TITLE
Adjust OpenGraph settings

### PR DESCRIPTION
**Pull request by amotl**
## What's inside

Mostly updates to the Open Graph configuration, using [sphinxext-opengraph](https://sphinxext-opengraph.readthedocs.io/), an extension to generate [Open Graph metadata](https://ogp.me/) for each documentation page.

- OpenGraph: Use new logo image.
- OpenGraph: Set description text length to 300 characters.
- OpenGraph: Disable use of "first image".
- Miscellaneous: 8a5521a just sorts the "Clients and Integrations" intersphinx link items alphabetically.

## Preview

Not much to see, a release will reveal it.

### CODE REVIEW

File: src/crate/theme/rtd/conf/__init__.py [hunk 1]
```diff
@@ -100,18 +100,19 @@
...
+    # CrateDB Clients and Integrations
     'crate-admin-ui': ('https://crate.io/docs/crate/admin-ui/en/latest/', None),
-    'crate-crash': ('https://crate.io/docs/crate/crash/en/latest/', None),
     'crate-clients-tools': ('https://crate.io/docs/crate/clients-tools/en/latest/', None),
+    'crate-crash': ('https://crate.io/docs/crate/crash/en/latest/', None),
+    'crate-dbal': ('https://crate.io/docs/dbal/en/latest/', None),
     'crate-jdbc': ('https://crate.io/docs/jdbc/en/latest/', None),
     'crate-npgsql': ('https://crate.io/docs/npgsql/en/latest/', None),
-    'crate-dbal': ('https://crate.io/docs/dbal/en/latest/', None),
+    'crate-operator': ('https://crate-operator.readthedocs.io/en/latest/', None),
```
Comment (msbt) [inline src/crate/theme/rtd/conf/__init__.py:115]: you sure about that URL?

**Review by msbt — CHANGES_REQUESTED** [review]

File: src/crate/theme/rtd/conf/__init__.py [hunk 2]
```diff
@@ -100,18 +100,19 @@
...
     'crate-tutorials': ('https://crate.io/docs/crate/tutorials/en/latest/', None),
     'crate-howtos': ('https://crate.io/docs/crate/howtos/en/latest/', None),
 
-    # CrateDB clients
+    # CrateDB Clients and Integrations
     'crate-admin-ui': ('https://crate.io/docs/crate/admin-ui/en/latest/', None),
-    'crate-crash': ('https://crate.io/docs/crate/crash/en/latest/', None),
     'crate-clients-tools': ('https://crate.io/docs/crate/clients-tools/en/latest/', None),
+    'crate-crash': ('https://crate.io/docs/crate/crash/en/latest/', None),
+    'crate-dbal': ('https://crate.io/docs/dbal/en/latest/', None),
```
Comment (amotl) [inline src/crate/theme/rtd/conf/__init__.py:112]: DBAL moved here.

File: src/crate/theme/rtd/conf/__init__.py (hunk 1 above)
Comment (amotl) [reply src/crate/theme/rtd/conf/__init__.py:115]: Ah, yes. Operator is not mounted to crate.io/docs.

File: src/crate/theme/rtd/conf/__init__.py [hunk 3]
```diff
@@ -128,12 +129,28 @@
...
+# - og:title 40 chars
+# - og:description has 2 max lengths:
+#   When the link is used in a Post, it's 300 chars. When a link is used in a Comment, it's 110 chars.
+#   So you can either treat it as 110, or, write your Descriptions to 300 but make sure the first 110
+#   is the critical part and still makes sense when it gets cut off.
+#   -- https://stackoverflow.com/questions/8914476/facebook-open-graph-meta-tags-maximum-content-length
+ogp_site_url = "https://crate.io/docs/"
+ogp_description_length = 300
+ogp_site_name = "CrateDB Documentation"
+ogp_image = "https://crate-docs-theme.readthedocs.io/en/latest/_static/images/cratedb-logo-h400.png"
```
Comment (msbt) [inline src/crate/theme/rtd/conf/__init__.py:151]: shouldn't we host that image on the other thing, so it a) comes from the crate.io domain and b) can replaced without editing the docs theme?

Comment (amotl) [reply src/crate/theme/rtd/conf/__init__.py:151]: It's not bad to have it self-contained either.

**Comment by amotl** [comment]
> Preview
> Not much to see, a release will reveal it.

I actually don't know why visiting the preview page https://crate-docs-theme--433.org.readthedocs.build/en/433/ does not display an Open Graph embed, for example within a Slack conversation.

This is the source code:

```html
<meta property="og:type" content="website" />
<meta property="og:url" content="https://crate.io/docs/python/index.html" />
<meta property="og:site_name" content="CrateDB documentation" />
<meta property="og:description" content="Table of contents Introduction, Documentation, Project information. Introduction: The Python client library for CrateDB implements the Python Datab..." />
<meta property="og:image" content="https://crate.io/hubfs/static/og_crateio.jpg" />
```

Maybe because `og:url`, which is https://crate.io/docs/python/index.html, responds with a 404?

**Comment by msbt** [comment]
@amotl that's because the image is at 
`https://crate-docs-theme--433.org.readthedocs.build/en/433/_static/images/cratedb-logo-h400.png`
and not
`https://crate-docs-theme.readthedocs.io/en/latest/_static/images/cratedb-logo-h400.png`
because it's built from a PR I reckon

**Comment by amotl** [comment]
The image URL is wrong, right. I was only asking why it does not display any embed at all.

**Review by msbt — APPROVED** [review]

**Pull request merged**